### PR TITLE
Remove dependency on libtensorflow_framework.so.

### PR DESCRIPTION
**Pull request by dan-zheng**
Remove `libtensorflow_framework.so` as a build product.
This should reduce toolchain size.

Follow-up attempt to https://github.com/apple/swift/pull/20898.

### CODE REVIEW

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please clean test tensorflow

File: utils/build_swift/driver_arguments.py [hunk 1]
```diff
@@ -1045,17 +1045,15 @@ def create_argument_parser():
     option('--tensorflow-host-lib-dir', store_path,
```
Comment (pschuh) [inline utils/build_swift/driver_arguments.py]: Technically, this host and target distinction is not needed anymore.

**Review by pschuh — APPROVED** [review]

File: utils/build_swift/driver_arguments.py (hunk 1 above)
Comment (dan-zheng) [reply utils/build_swift/driver_arguments.py]: Some context: I recall we added the TensorFlow host/target distinction to facilitate future cross-compilation support, at @gribozavr's recommendation.

We mimicked the host/target distinction for some ICU flags (Android-related?) but I can't seem to find those anymore.

Comment (pschuh) [reply utils/build_swift/driver_arguments.py]: Yes, but we no longer compile it into swiftc, so the target one is always for cross-compilation.

Comment (dan-zheng) [reply utils/build_swift/driver_arguments.py]: That makes sense!

**Comment by dan-zheng** [comment]
Please hold off on merging until toolchain building is tested locally.
I'm testing macOS toolchain building locally - can't test Ubuntu toolchain building since I'm currently remote.

**Comment by dan-zheng** [comment]
Toolchain building succeeded on macOS.
Exact toolchain size decrease is not clear because I tested with another WIP patch - will post an update on my next patch.

**Pull request merged**